### PR TITLE
Update CI to include integration test

### PR DESCRIPTION
**Pull request by CtrlSpice**
This test runs a bash script that 
- launches the executable in the background 
- sends a minimal test trace using a curl command
- queries the trace store via api call to make sure the trace data was received and processed correctly
- kills the running process and exits with the appropriate exit code

Note: I have added a "CI" environment variable to prevent the browser window from being launched when running in CI. This is set in `go.yaml` under the Integration step.

### CODE REVIEW

File: .github/workflows/integration.sh [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+if [ $rootServiceName == '"test-with-curl"' ]
+then
+    echo 'Exit status 0: All good.'
+    kill -15 $pid
+    exit 0
+else
+    echo 'Exit status 1: unexpected rootServiceName ' + $rootServiceName
+    kill -15 $pid
+    exit 1
+fi
```
Comment (jmorrell) [inline .github/workflows/integration.sh:31]: As far as bash scripts go, this is really clean and nice 💯

**Review by jmorrell — APPROVED** [review]
DALL·E 2023-03-02 13 22 54 - a corgi sad about debugging a program, digital art

File: .github/workflows/integration.sh (hunk 1 above)
Comment (CtrlSpice) [reply .github/workflows/integration.sh:31]: Really?! I guess as far as bash scripts go... 🤣

File: .github/workflows/go.yml [hunk 2]
```diff
@@ -8,7 +8,11 @@ on: push
 jobs:
 
   build:
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
```
[the remark this answers is not in the document]
Comment (CtrlSpice) [reply .github/workflows/go.yml:14]: And they pass, too! 😁

**Pull request merged**